app/controllers/concerns/filter_parameters.rb

```diff
@@ -0,0 +1,12 @@
+module FilterParameters
+private
+
+  def filter_params
+    params_to_pass = params.reject do |param|
+      %w(authenticity_token utf8 action controller).include?(param)
+    end
+
+    allowed_arrays = params_to_pass.keys.each_with_object({}) { |key, hash| hash[key] = [] }
+    params.permit(*params_to_pass.keys, allowed_arrays)
```

Comment:
Why not use `request.request_parameters`?

Reply:
We need to also add the params from the filter form in here so... query_params + form params. I think we have to use param in that case if we're going to reuse this for all filter pages.

Reply:
Totally misread that as`query_parameters` not `request_parameters`. Yes, that would be loads simpler.

Reply:
Fixed in 4372712. `utf8` was still in there. I've filtered it out just because it isn't there in the existing implementation and I don't know what search and compare will do with it. Probably nothing 🤷‍♂